Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -1038,6 +1038,64 @@ void StyleComputer::cascade_declarations(
     }
 }
 
+// https://html.spec.whatwg.org/multipage/rendering.html#margin-collapsing-quirks
+void StyleComputer::apply_margin_collapsing_quirk(CascadedProperties& cascaded_properties, DOM::Element& element) const
+{
+    // In quirks mode,
+    if (!document().in_quirks_mode())
+        return;
+
+    bool add_margin_block_start_zero_rule = false;
+    bool add_margin_block_end_zero_rule = false;
+    // any element with default margins
+    if (element.is_element_with_default_margins()) {
```

Comment:
Disclaimer: I don't know much about this CSS implementation, but invalidation doesn't seem to be handled.

Reply:
Good point, I had not considered that.
Though, at this point, I'm not too sure about this way of implementing the quirk as a whole anymore.
The Chromium way of doing it seems more web compatible and when I wrote all this, I still had more of a "if it says so in the spec, it must be right" take on the whole thing, since I didn't know how that bit of spec came to be.